### PR TITLE
Json adapter from json implementation test

### PR DESCRIPTION
**Pull request by rafbel**

### CODE REVIEW

**Comment by ZacSweers** [comment]
Hi, the lack of context makes it impossible for us to review these. Can you please give a proper description?

File: moshi/pom.xml [hunk 1]
```diff
@@ -32,6 +32,12 @@
       <artifactId>assertj-core</artifactId>
       <scope>test</scope>
     </dependency>
+      <dependency>
+          <groupId>org.junit.jupiter</groupId>
+          <artifactId>junit-jupiter</artifactId>
```
Comment (JakeWharton) [inline moshi/pom.xml]: Stick to JUnit 4, please.

File: moshi/src/main/java/com/squareup/moshi/JsonAdapter.java [hunk 2]
```diff
@@ -97,6 +97,18 @@ public final void toJson(BufferedSink sink, @Nullable T value) throws IOExceptio
     }
   }
 
+  /**
+   * Decodes a Java value object from {@code value}, which must be comprised of maps, lists,
+   * strings, numbers, booleans and nulls.
```
Comment (JakeWharton) [inline moshi/src/main/java/com/squareup/moshi/JsonAdapter.java]: Javadoc is wrong. Copied from elsewhere?

File: moshi/src/main/java/com/squareup/moshi/internal/Util.java [hunk 3]
```diff
@@ -636,4 +636,5 @@ public static JsonDataException unexpectedNull(
     }
     return new JsonDataException(message);
   }
+
```
Comment (JakeWharton) [inline moshi/src/main/java/com/squareup/moshi/internal/Util.java]: Revert

File: moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    public void initializeBeforeEach() {
+        this.moshi = new Moshi.Builder().build();
+    }
+
+    @Test
+    public void nonNullJsonAdapterTest() throws IOException {
+        String json = "null";
+
+        JsonAdapter<Person> adapter = moshi.adapter(Person.class);
+        boolean exceptionThrown = false;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java]: Remove and put a `fail()` inside the `try`.

File: moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    @Test
+    public void nullSafeToNonNullJsonAdapterTest() throws IOException {
+        String json = "null";
+
+        JsonAdapter<Person> adapter = moshi.adapter(Person.class).nullSafe();
+
+        assertNull(adapter.fromJson(json));
+        boolean exceptionThrown = false;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java]: Same

File: moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        Person testPerson = adapter.lenient().fromJson(json);
+        assertThat(testPerson.money).isEqualTo(Double.POSITIVE_INFINITY);
+    }
+
+    @Test
+    public void notLenientTest() throws IOException {
+        String json = "{\"name\": \"Rafael\", \"age\": 24, \"money\": Infinity}";
+
+        JsonAdapter<Person> adapter = moshi.adapter(Person.class);
+        boolean exceptionThrown = false;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java]: Same

File: moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        assertEquals(adapter.serializeNulls().toJson(samplePerson), sampleJson);
+    }
+
+    @Test
+    public void failOnUnknownJsonAdapterTest() throws Exception {
+
+        String sampleJson = "{\"name\": \"Rafael\", \"ag\": 24, \"money\": 3.8}";
+
+        JsonAdapter<Person> adapter = moshi.adapter(Person.class);
+        boolean assertOccurred = false;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java]: Same

File: moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+        assertTrue(assertOccurred);
+    }
+
+    @Test
+    public void failOnUnknownToFailOnUnknownJsonAdapterTest() throws Exception {
+        String sampleJson = "{\"name\": \"Rafael\", \"ag\": 24, \"money\": 3.8}";
+
+        JsonAdapter<Person> adapter = moshi.adapter(Person.class).failOnUnknown();
+
+        boolean assertOccurred = false;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java]: Same

File: moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+        assertTrue(assertOccurred);
+    }
+
+    @Test
+    public void notFailOnUnknownTest() throws IOException {
+        String sampleJson = "{\"name\": \"Rafael\", \"ag\": 24, \"money\": 3.8}";
+
+        JsonAdapter<Person> adapter = moshi.adapter(Person.class);
+
+        boolean assertOccurred = false;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/JsonAdapterStateTest.java]: Same

File: moshi/src/test/java/com/squareup/moshi/TypesTest.java [hunk 10]
```diff
@@ -347,6 +348,18 @@ public static Type getFirstTypeArgument(Type type) throws Exception {
     assertThat(adapter).isNotNull();
   }
 
+  @Test public void getRawTypeThrowsException() {
+    Type type = null;
+    Boolean exceptionThrown = false;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/TypesTest.java]: Remove and use `fail()`.

File: moshi/src/test/java/com/squareup/moshi/TypesTest.java [hunk 11]
```diff
@@ -347,6 +348,18 @@ public static Type getFirstTypeArgument(Type type) throws Exception {
     assertThat(adapter).isNotNull();
   }
 
+  @Test public void getRawTypeThrowsException() {
+    Type type = null;
```
Comment (JakeWharton) [inline moshi/src/test/java/com/squareup/moshi/TypesTest.java]: Can be inlined

File: pom.xml [hunk 12]
```diff
@@ -4,203 +4,70 @@
...
   <parent>
-    <groupId>org.sonatype.oss</groupId>
-    <artifactId>oss-parent</artifactId>
-    <version>7</version>
+    <groupId>com.squareup.moshi</groupId>
+    <artifactId>moshi-parent</artifactId>
+    <version>1.10.0-SNAPSHOT</version>
   </parent>
 
-  <groupId>com.squareup.moshi</groupId>
```
Comment (JakeWharton) [inline pom.xml]: Revert all this

**Comment by rafbel** [comment]
Sorry, I ended up creating a pull request using the wrong branches. This was for study purposes only. I will close this.

**Pull request closed without merging**